io.sloeber.ui/src/io/sloeber/ui/monitor/views/SerialMonitor.java

```diff
@@ -511,25 +513,40 @@ void SerialPortsUpdated() {
 
 	/**
 	 * Connect to a serial port and sets the listener
-	 * 
+	 *
 	 * @param comPort
 	 *            the name of the com port to connect to
 	 * @param baudRate
 	 *            the baud rate to connect to the com port
 	 */
 	public void connectSerial(String comPort, int baudRate) {
 		if (serialConnections.size() < MY_MAX_SERIAL_PORTS) {
-			int colorindex = serialConnections.size();
+			int colorindex = -1;
+			for (int idx = 0; idx < serialPortAllocated.length; idx++) {
+				if ( ! serialPortAllocated[idx]) {
+					colorindex = idx;
+					break;
+				}
+			}
+
+			if (colorindex < 0) {
+				Activator.log(new Status(IStatus.ERROR, Activator.getId(), Messages.serialMonitorNoMoreSerialPortsSupported,
+						null));
+			}
+
 			Serial newSerial = new Serial(comPort, baudRate);
 			if (newSerial.IsConnected()) {
 				newSerial.registerService();
 				SerialListener theListener = new SerialListener(this, colorindex);
 				newSerial.addListener(theListener);
-				String newLine=System.getProperty("line.separator");//$NON-NLS-1$
```

Comment:
To me newLine is far more readable then NL. So I'm not sure what the added value is of using NL

Reply:
I should have named the field NEW_LINE. But it's an unrelated change so I'll just undo it.

Reply:
As you may have noticed I like verbose code ;-) NEW_LINE is ok for me